src/Symfony/Bridge/PhpUnit/ClockMock.php

```diff
@@ -74,9 +74,10 @@ public static function register($class)
         $self = get_called_class();
 
         $mockedNs = array(substr($class, 0, strrpos($class, '\\')));
-        if (strpos($class, '\\Tests\\')) {
-            $ns = str_replace('\\Tests\\', '\\', $class);
-            $mockedNs[] = substr($ns, 0, strrpos($ns, '\\'));
+        $count = 0;
+        $ns = preg_replace('/(^|\\\\)Tests\\\\/', '$1', $mockedNs[0], 1, $count);
+        if (1 === $count) {
+            $mockedNs[] = $ns;
```

Comment:
Using regex seems more extensible to me, even though it might not be as readable...